backend/fms_core/services/dataset.py

```diff
@@ -0,0 +1,65 @@
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+
+from datetime import datetime, date
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from fms_core.models.dataset_file import DatasetFile
+from fms_core.models.dataset import Dataset
+from fms_core.models._constants import ReleaseFlag
+
+def create_dataset(project_name: str, run_name: str, lane: str, files: List[Dict[str, Any]] = []) -> Tuple[Union[Dataset, None], List[str], List[str]]:
+    dataset = None
+
+    errors = []
+    warnings = []
+
+    try:
+        _ = int(lane)
+        dataset = Dataset.objects.create(
+            project_name=project_name,
+            run_name=run_name,
+            lane=lane,
+        )
+    except ValidationError as e:
+        errors.extend(e.messages)
+    except ValueError as e:
+        errors.append(f"Lane is '{lane}' which is not an integer")
+    
+    
+    dataset_files = []
+
+    for file in files:
+        if not errors:
+            dataset_file, newerrors, newwarnings = create_dataset_file(dataset=dataset, **file)
+            errors.extend(newerrors)
+            warnings.extend(newwarnings)
+
+            if dataset_file:
+                dataset_files.append(dataset_file)
+
+        dataset = None
+        dataset_files = []
+
+    return (dataset and Dataset.objects.get(pk=dataset.id), errors, warnings)
+
+def create_dataset_file(dataset: Dataset,
+                        file_path: str,
+                        sample_name: str,
+                        release_flag: int = ReleaseFlag.BLOCK
+                        ) -> Tuple[Union[DatasetFile, None], List[str], List[str]]:
+    dataset_file = None
+    errors = []
+    warnings = []
+
+    try:
+        dataset_file = DatasetFile.objects.create(
+            dataset=dataset,
+            file_path=file_path,
+            sample_name=sample_name,
```

Comment:
Should the service check if a sample with the given sample_name exists in the db, or is the name just informative?

Reply:
For now it is informative.